android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -399,10 +400,12 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
             onConfirm("")
         }
         val initialDateString = prompt.defaultValue ?: ""
-        val stepValue = if (prompt.stepValue.isNullOrBlank()) {
-            null
-        } else {
-            prompt.stepValue
+        val stepValue = with(prompt.stepValue) {
+            if (this.isNullOrBlank() || !this.isDigitsOnly()) {
```

Comment:
There is a failing test - `onDateTimePrompt DATETIME_TYPE_TIME with stepValue time parameter must format time correctly` for the scenario in which the step is `0.1`that seems to be affected by the `isDigitsOnly` check - because of the dot character.
We could probably use a `prompt.stepValue?.toDoubleOrNull() == null` for the check and avoid such issues.
(Great that you added good tests beforehand!)